libvast/src/system/index.cpp

```diff
@@ -853,8 +862,12 @@ index(index_actor::stateful_pointer<index_state> self,
       auto synopsis_path = self->state.partition_synopsis_path(partition_id);
       bool adjust_stats = true;
       if (self->state.persisted_partitions.count(partition_id) == 0u) {
-        if (!exists(path)) {
-          rp.deliver(caf::make_error(ec::logic_error, "unknown partition"));
+        std::error_code err{};
+        const auto file_exists = std::filesystem::exists(path, err);
+        if (!file_exists) {
```

Comment:
Error should also be checked here.

Reply:
See https://github.com/tenzir/vast/pull/1504#discussion_r602732512